Source/SwiftLintFramework/Rules/UnusedOptionalBindingRule.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  UnusedOptionalBindingRule.swift
+//  SwiftLint
+//
+//  Created by Rafael Machado on 1/5/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct UnusedOptionalBindingRule: ASTRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "unused_optional_binding",
+        name: "Unused Optional Binding",
+        description: "Prefer `!= nil` over `let _ =`",
+        nonTriggeringExamples: [
+            "if let bar = Foo.optionalValue {\n" +
+            "}\n",
+            "if let (_, second) = getOptionalTuple() {\n" +
+            "}\n"
+        ],
+        triggeringExamples: [
+            "if let ↓_ = Foo.optionalValue {\n" +
+            "}\n",
+            "if let a = Foo.optionalValue, let ↓_ = Foo.optionalValue2 {\n" +
+            "}\n",
+            "guard let a = Foo.optionalValue, let ↓_ = Foo.optionalValue2 {\n" +
+            "}\n",
+            "if let (first, second) = getOptionalTuple(), let ↓_ = Foo.optionalValue {\n" +
+            "}\n",
+            "if let (first, _) = getOptionalTuple(), let ↓_ = Foo.optionalValue {\n" +
+            "}\n",
+            "if let (_, _) = getOptionalTuple(), let ↓_ = Foo.optionalValue {\n" +
```

Comment:
@jpsim good point! Haven't thought of it...Any suggestions on how to properly validate tuples?